### PR TITLE
Upgrade to Reason V3

### PR DESCRIPTION
**Pull request by chenglou**

### CODE REVIEW

**Comment by chenglou** [comment]
@glennsl no output difference after https://github.com/reasonml-community/bs-webapi-incubator/commit/47bdcbb91e158666882652ef0b2de126469b27ce

I'll let you decide if the output looks good enough. If not, tell me what you'd like to see manually tweaked

**Comment by glennsl** [comment]
Oh man, that would make a rather large list. I'll sort it out though.

File: src/canvas/Canvas2dRe.re [hunk 1]
```diff
@@ -37,19 +36,33 @@ module Composite: CompositeType = {
   let xor: t = "xor";
 };
 
-module type LineCapType = {type t = pri string; let butt: t; let round: t; let square: t;};
```
Comment (glennsl) [inline src/canvas/Canvas2dRe.re]: I'm not sure toplevel module definitions should be formatted into one-liners.

File: src/dom/css/CssStyleDeclarationRe.re [hunk 2]
```diff
@@ -1,267 +1,138 @@
...
 [@bs.get] external borderBottom : t => string = "";
-
 [@bs.get] external borderLeft : t => string = "";
-
 [@bs.get] external borderTopColor : t => string = "";
-
 [@bs.get] external borderRightColor : t => string = "";
-
 [@bs.get] external borderBottomColor : t => string = "";
-
```
Comment (glennsl) [inline src/dom/css/CssStyleDeclarationRe.re]: There were a couple thousand too many blank lines I feel.

File: src/dom/html/HtmlElementRe.re [hunk 3]
```diff
@@ -10,45 +11,42 @@ module Impl = (T: {type t;}) => {
...
     (self, value) => setDir(self, DomTypesRe.encodeDir(value));
   [@bs.get] external draggable : t_htmlElement => bool = "";
-  [@bs.set] external setDraggable : (t_htmlElement, Js.boolean) => unit =
-    "draggable" /* temproarily removed to reduce codegen size */; /*let setDraggable : t_htmlElement => bool => unit = fun self value => setDraggable self (Js.Boolean.to_js_boolean value);*/
+  [@bs.set] external setDraggable : (t_htmlElement, Js.boolean) => unit = "draggable";
+  /*let setDraggable : t_htmlElement => bool => unit = fun self value => setDraggable self (Js.Boolean.to_js_boolean value);*/ /* temproarily removed to reduce codegen size */
   [@bs.get] external dropzone : t_htmlElement => Dom.domSettableTokenList = "";
   [@bs.get] external hidden : t_htmlElement => bool = "";
-  [@bs.set] external setHidden : (t_htmlElement, Js.boolean) => unit =
-    "hidden" /* temproarily removed to reduce codegen size */; /*let setHidden : t_htmlElement => bool => unit = fun self value => setHidden self (Js.Boolean.to_js_boolean value);*/
```
Comment (glennsl) [inline src/dom/html/HtmlElementRe.re]: Comments get switched around when there are multiple on the same line.

File: src/dom/nodes/CdataSectionRe.re [hunk 4]
```diff
@@ -1,43 +1,8 @@
...
-  );
-
-include
-  NonDocumentTypeChildNodeRe.Impl(
-    {
-      type nonrec t = t;
-    }
-  );
-
-include
```
Comment (glennsl) [inline src/dom/nodes/CdataSectionRe.re]: While toplevel modules get one-lined, One-line functor arguments apparently need to be exploded into six lines, lol.

**Pull request merged**